tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -812,6 +813,9 @@ private Map<TopicPartition, LogOffsetResult> getLogOffsets(Collection<TopicParti
                         : new Unknown()
                 ));
             } catch (InterruptedException | ExecutionException e) {
+                if (e.getMessage().contains(UnknownTopicOrPartitionException.class.getCanonicalName())) {
```

Comment:
This seems better as `if (e.getCause() instanceof UnknownTopicOrPartitionException)`.

Reply:
Also, I think you need similar logic in `getLogTimestampOffsets` since that's used in a similar way for a different kind of resetting offsets and I think it would exhibit the same problem you're trying to fix.

Reply:
`getLogTimestampOffsets` I am not sure we should target reset offset as it only impact individual group/topic metadata and as far as I know you can't reset offsets for all groups.

Reply:
The main goal of this jira and bug is if one topic is deleted describing full metadata of cluster or the rest of the group in case of multiple topic subscription shouldn't fail

Reply:
Looking into reset offset test and apparently the script can also use `--all-groups` and `--all-topics` which will hit the same issue if any single group has a deleted topic in any of the combinations that has any of these fields. I'll add tests to reset offset to catch this and fix it.

Reply:
Ah, OK. I didn't realise the specific situation you were fixing. There is a window in reset-offsets but it's hard to hit and I now understand it's out of scope. Thanks for the PR.